src/EditorFeatures/CSharpTest/CodeActions/ConvertConversionOperators/ConvertConversionOperatorsFromCastTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.CSharpCodeRefactoringVerifier<Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertConversionOperators.CSharpConvertConversionOperatorsFromCastRefactoringProvider>;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.ConvertConversionOperators
+{
+    [Trait(Traits.Feature, Traits.Features.ConvertConversionOperators)]
+    public class ConvertConversionOperatorsFromCastTests
+    {
+        [Fact]
+        public async Task ConvertFromExplicitToAs()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = ([||]object)1;
+    }
+}";
+            const string ExpectedMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as object;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = ExpectedMarkup,
+                CodeActionValidationMode = CodeActionValidationMode.Full,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromExplicitToAs_ValueType()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = ([||]byte)1;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = InitialMarkup,
+                OffersEmptyRefactoring = false,
+                CodeActionValidationMode = CodeActionValidationMode.None,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromExplicitToAs_ValueTypeConstraint()
```

Comment:
TODO: Add unconstraint test:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#the-as-operator
```CS
    public U H<U>(object o) {
        return o as U;             // Error, U is unconstrained 
    }
```

Reply:
Type parameter handling was wrong. Fixed it in 
f18b28d and moved the filter logic to a common extension method for VB and C#.